platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/NoOpMerkleStateLifecycles.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.util;
+
+import com.hedera.hapi.block.stream.output.StateChanges;
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.platform.state.MerkleStateLifecycles;
+import com.swirlds.platform.state.MerkleStateRoot;
+import com.swirlds.platform.state.service.PlatformStateService;
+import com.swirlds.platform.state.service.schemas.V0540PlatformStateSchema;
+import com.swirlds.platform.system.*;
+import com.swirlds.platform.system.address.AddressBook;
+import com.swirlds.platform.system.events.Event;
+import com.swirlds.state.State;
+import com.swirlds.state.lifecycle.MigrationContext;
+import com.swirlds.state.lifecycle.StateDefinition;
+import com.swirlds.state.lifecycle.info.NetworkInfo;
+import com.swirlds.state.merkle.StateMetadata;
+import com.swirlds.state.merkle.singleton.SingletonNode;
+import com.swirlds.state.spi.*;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.MarkerManager;
+
+public enum NoOpMerkleStateLifecycles implements MerkleStateLifecycles {
+    NO_OP_MERKLE_STATE_LIFECYCLES;
+
+    private static final Logger logger = LogManager.getLogger(NoOpMerkleStateLifecycles.class);
+
+    public List<StateChanges.Builder> initPlatformState(@NonNull final State state) {
+        logger.info(MarkerManager.getMarker("DEMO_INFO"), "Init Platform State...");
+
+        if (!(state instanceof MerkleStateRoot merkleStateRoot)) {
+            throw new IllegalArgumentException("Can only be used with MerkleStateRoot instances");
+        }
+        final var schema = new V0540PlatformStateSchema();
+        schema.statesToCreate().stream()
+                .sorted(Comparator.comparing(StateDefinition::stateKey))
```

Comment:
I see that this code is very similar to what we're having in the `FakeMerkleStateLifecycles` in test fixtures.

I suggest to find a way to extract the common code into a single class instead of replicating it. This will ensure the consistent behavior between demo apps and tests, and wherever else we happen to use this functionality.

Reply:
Yes, it's a good idea. So, I got rid of this class and used `FakeMerkleStateLifecycles`. It caused some trouble with the build, but thanks to @jjohannes it's resolved now.